Wrappers/Python/cil/optimisation/functions/TotalVariation.py

```diff
@@ -274,11 +286,19 @@ def _fista_on_dual_rof(self, x, tau, out = None):
             self.calculate_Lipschitz()            
         
         # initialise
-        t = 1        
+        t = 1        # line 2 
+
+        p1 = self.gradient.range_geometry().allocate(0) # dua variable - value alloacated here is not used -  is overwritten during iterations 
+        if not self.warmstart:
+            self.p2 = self.gradient.range_geometry().allocate(0)  # previous dual variable - needs loading for warm start! 
+            tmp_q = self.gradient.range_geometry().allocate(0) # should be equal to self.p2 - i.e. needs loading for warm start 
+        else:
+            if not self.hasstarted:
+                self.p2 = self.gradient.range_geometry().allocate(0)  # previous dual variable - needs loading for warm start! 
+                self.hasstarted = True
+            tmp_q = self.p2.copy() # should be equal to self.p2 - i.e. needs loading for warm start 
+
```

Comment:
this could be simplified as 

```suggestion
        t = 1        # line 2 
        # dual variable - its content is overwritten during iterations 
        p1 = self.gradient.range_geometry().allocate(None)
        # should be equal to self.p2 - i.e. needs loading for warm start
        tmp_q = self.p2.copy()  

```

Reply:
Thanks - have changed this!